File: 백준/Gold/2447. 별 찍기 － 10/별 찍기 － 10.py
import sys, math
input = sys.stdin.readline

N = int(input())
arr = [["*"] * N for _ in range(N)]

def star(x, start, end, N):
    if N >= 3:
        for i in range(start+N//3, (start+N//3)+N//3):
            for j in range(end+N//3, (end + N//3)+N//3):
                x[i][j] = " "

        for n in range(start, start+N, N//3):
            for m in range(end, end+N, N//3):
                star(x, n, m, N//3)

star(arr, 0, 0, N)
for i in arr:
    print("".join(i))

File: 백준/Gold/2447. 별 찍기 － 10/별 찍기 － 10.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim()

const N = parseInt(input)
const arr = Array.from({ length: N }, () => Array(N).fill('*'));

function star(x, start, end, N) {
    if (N >= 3) { 
        for (let i = start+parseInt(N/3); i < (start + 2 * parseInt(N/3)); i++) {
            for (let j  = end+parseInt(N/3); j < (end + 2 * parseInt(N/3)); j++) {
                x[i][j] = " "
            }
        }
        
        for (let n = start; n < start + N; n += parseInt(N/3)) {
            for (let m = end; m < end + N; m += parseInt(N/3)) {
                star(x, n, m, parseInt(N/3))
            }
        }
    }
}

star(arr, 0, 0, N)
let answer = ''
arr.forEach(x => {
    answer += x.join("") + "\n"

})
console.log(answer)